### PR TITLE
Tesla Corona Analyzers

### PR DESCRIPTION
**Pull request by Poojawa**
:cl: Poojawa
add: Tesla Corona Analyzers! Study the seemingly magic Edison's Bane for supplemental research points!
/:cl:

Pubby does not come with an Supermatter engine, thus cannot produce tritium for fusion warheads for science. This would also permit a return of Tesla Engines for other uses as well, if/when a point generation nerf is implemented.

The maximum point cap and resistance values are just randomly assigned, and should probably be tweaked as needed. Using the research setting induces a 20 (up from 2) Power_loss value, effectively making voltages from 120,000W strikes to 6,000W, but there is no cap to power generation in this setting. Points per Tesla_act are hard capped at 200, but like I said I pulled it out of my ass assuming that having multiple energy balls doesn't just call tesla_act multiple times at once, rather than just adding to the power generation as a whole. 

Power setting is like normal, with the added small bonus of points to reflect the focus of hard power transfer.  You can swap between modes as you would the gas freezer/heater, as I found that the best alternative to trying to hack in a mod kit or multitool fuckery. 

Currently none of the maps are edited for specific machines, so they'll all default to power mode unless deconstructed and swapped over. 

Proof of working:
`ADMIN LOG: Poojawa/(Flick Kelley) called Tesla Corona Analyzer's tesla_act() with the arguments 200.
Available points: 10 (+0 / minute)`

`ADMIN LOG: Poojawa/(Flick Kelley) called Tesla Corona Analyzer's tesla_act() with the arguments 547387.
Available points: 210 (+0 / minute)`

Sprites! 

zap fixed

daea49e50ab37cfff413aaaa3056cf24

### CODE REVIEW

File: code/modules/power/tesla/coil.dm [hunk 1]
```diff
@@ -91,6 +99,30 @@
...
+/obj/machinery/power/tesla_coil/research/tesla_act(var/power)
+	if(anchored && !panel_open)
+		obj_flags |= BEING_SHOCKED
+		//don't lose arc power when it's not connected to anything
+		//please place tesla coils all around the station to maximize effectiveness
+		var/power_produced = powernet ? power / power_loss : power
+		add_avail(power_produced*input_power_multiplier)
+		flick("coilhit", src)
+		playsound(src.loc, 'sound/magic/lightningshock.ogg', 100, 1, extrarange = 5)
+		tesla_zap(src, 5, power_produced)
```
[the remark this answers is not in the document]
Comment (Poojawa) [reply code/modules/power/tesla/coil.dm]: on the sound or ...?

**Comment by Poojawa** [comment]
Just tested it, tesla_act is only called once. per strike.

**Comment by DaedalusGame** [comment]
(also i just read the original description, tritium can almost certainly be made on pubby and used both, for radiation collectors or bombs, you make tritium in a turbine engine or burn chamber with minor modifications, not in the supermatter engine)

**Comment by Poojawa** [comment]
Yah, but it's obviously easier/more convenient to use the SME. This is also just convenience if someone wants to play with Tesla.

**Comment by ExcessiveUseOfCobblestone** [comment]
It would be more consistent to just have this as part of the Tesla coil an you can toggle between power gen and research.

This is neat

**Comment by DaedalusGame** [comment]
t-thats what this is tho
but yeah it'd be better if it was toggleable like rad collectors with a multitool

**Comment by Poojawa** [comment]
Updated with some fairly basic, but useful sprites.

**Comment by Poojawa** [comment]
oh no. Sorry my dudes.

**Comment by WJohn** [comment]
The sprites are... pretty darned lazy. They're just tesla coils with a metal cap on top of them. At least try to work on the style of the original sprite and make one of your own. They're fairly simple, all things considered. No directionals, a tooon of repeating textures, and pretty limited color palette (6 greys, 3 pinks).

See if you can't come up with a more interesting variant than this. You can re-use the electrical arcs, those are very complicated and there's no real need for new ones.

**Review by WJohn — CHANGES_REQUESTED** [review]
Check comment etc. I gotta put my reviews in the review summary from now on...

**Comment by Poojawa** [comment]
To be honest, the sprites were just stopgap just in case someone wanted a visual reference for the mode. I figured it'd be nice to have them as an option instead of flat out changing the entire machine sprite.

**Comment by ExcessiveUseOfCobblestone** [comment]
@DaedalusGame one uses multitool while the other requires you dismantle the machine for the circuit and minor differences are not acceptable!!!!

**Comment by DaedalusGame** [comment]
ITT Cobby reads half of my comment as usual

**Comment by ExcessiveUseOfCobblestone** [comment]
I was explaining my comment (1st half of urs) in agreement with your 2nd half ni:b::b:a

**Comment by Poojawa** [comment]
new zaaap

daea49e50ab37cfff413aaaa3056cf24


How are these?

**Comment by WJohn** [comment]
Much better! I am however noticing a frame where the lights turn off when it is struck, which the original tesla coil doesn't do, and a few sparks of lightning you've created there have a color that's way, way off the original (set edit mode to rgb + mask and then use the swap (W) tool to pick the right color, delete the coil, then copy paste your cleaned up lightning effect on top of a clear coil (with lights on)).

If you wanna go the extra mile, since these already look nearly identical to the original tesla coil, you could basically make this a different MODE to them. Rather than collecting power, they collect science research when toggled to this, so you can work a tradeoff mechanic into it.

**Comment by Poojawa** [comment]
Uhh. That's what they already do. It's a swapped board mode between power (coils already here) and research at the cost of power production  (new sprites). 

I'll admit that the lightning effects are a bit tough, but the alpha on all the lightning was the same 120-141 or so. The corona effect in real applications is stated to be a blue-ish or relative glow around the high voltage item. 

Also the Tesla coil's last frame is a resetted one, but the animation also way better matches it. Since it's only a flick, it repeating make it look weird in the .gif

**Comment by cebutris** [comment]
You shouldn't have to disassemble them in order to swap modes, that would make it nearly impossible to do if the tesla is already running- if you finalize the construction but don't secure it immediatley after the tesla will blow you up

**Comment by WJohn** [comment]
You know what, this is acceptable. I don't want to hold up this PR any longer.

**Review by WJohn — APPROVED** [review]

**Comment by DaedalusGame** [comment]
@cebutris i have a pr up to hopefully fix said problem (#35968)

**Pull request merged**